### PR TITLE
Factor out manifest requests

### PR DESCRIPTION
**Pull request by andresuribe87**
# Overview
This provides a similar mechanism to the presentation requests pattern. 

# Description
I moved shared logic to a common folder. Open to name suggestions. 

# How Has This Been Tested?
Added unit tests, and modified existing ones where it made sense.

### CODE REVIEW

File: pkg/server/router/manifest.go [hunk 1]
```diff
@@ -573,3 +572,173 @@ func (mr ManifestRouter) ReviewApplication(c *gin.Context) {
...
+//	@Produce		json
+//	@Param			id	path		string	true	"ID"
+//	@Success		204	{string}	string	"No Content"
+//	@Failure		400	{string}	string	"Bad request"
+//	@Failure		500	{string}	string	"Internal server error"
+//	@Router			/v1/manifests/requests/{id} [delete]
+func (mr ManifestRouter) DeleteRequest(c *gin.Context) {
+	id := framework.GetParam(c, IDParam)
+	if id == nil {
+		errMsg := "cannot delete a manigest request without an ID parameter"
```
Comment (decentralgabe) [inline pkg/server/router/manifest.go:711]:
```suggestion
		errMsg := "cannot delete a manifest request without an ID parameter"
```

File: pkg/server/router/presentation.go [hunk 2]
```diff
@@ -457,7 +456,7 @@ func (pr PresentationRouter) ReviewSubmission(c *gin.Context) {
 	framework.Respond(c, ReviewSubmissionResponse{Submission: submission}, http.StatusOK)
 }
 
-type CreateRequestRequest struct {
+type CommonCreateRequestRequest struct {
```
Comment (decentralgabe) [inline pkg/server/router/presentation.go:459]: perhaps move this into a `model.go`?

File: pkg/service/common/request.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	return &Request{
+		ID:         stored.ID,
+		Audience:   stored.Audience,
+		Expiration: expiration,
+		IssuerDID:  stored.IssuerDID,
+		IssuerKID:  stored.IssuerKID,
+	}, nil
+}
+
+func CreateStoredRequest(
```
Comment (decentralgabe) [inline pkg/service/common/request.go:47]: nit: prefer this to be a single line with a single line break if needed

File: pkg/service/common/request.go [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	ctx context.Context,
+	keyStore *keystore.Service,
+	claimName string,
+	claimValue any,
+	request Request,
+	id string,
+) (StoredRequest, error) {
```
Comment (decentralgabe) [inline pkg/service/common/request.go:54]:
```suggestion
) (*StoredRequest, error) {
```

so we can return nil, err

File: pkg/service/common/storage.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+	IssuerKID   string   `json:"issuerKid"`
+	ReferenceID string   `json:"referenceId"`
+	JWT         string   `json:"jwt"`
+}
+
+type RequestStorage interface {
+	StoreRequest(context.Context, StoredRequest) error
+	GetRequest(context.Context, string) (*StoredRequest, error)
+	DeleteRequest(context.Context, string) error
+	ListRequests(context.Context) ([]StoredRequest, error)
```
Comment (decentralgabe) [inline pkg/service/common/storage.go:26]:
```suggestion
	DeleteRequest(context.Context, string) error
	ListRequests(context.Context) ([]StoredRequest, error)
```

nit: ordering

File: pkg/service/common/storage.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+}
+
+type requestStorage struct {
```
Comment (decentralgabe) [inline pkg/service/common/storage.go:29]: I like that this isn't exported - we should it this for other storage too

File: pkg/service/manifest/model/model.go [hunk 7]
```diff
@@ -14,19 +15,19 @@ import (
...
-	OutputDescriptors      []manifestsdk.OutputDescriptor   `json:"outputDescriptors" validate:"required,dive"`
-	ClaimFormat            *exchange.ClaimFormat            `json:"format" validate:"required,dive"`
+	Name              *string                        `json:"name,omitempty"`
+	Description       *string                        `json:"description,omitempty"`
+	IssuerDID         string                         `json:"issuerDid" validate:"required"`
+	IssuerKID         string                         `json:"issuerKid" validate:"required"`
+	IssuerName        *string                        `json:"issuerName,omitempty"`
+	OutputDescriptors []manifestsdk.OutputDescriptor `json:"outputDescriptors" validate:"required,dive"`
+	ClaimFormat       *exchange.ClaimFormat          `json:"format" validate:"required,dive"`
+	// TODO: Allow for specifying the presentation definition only by id.
```
Comment (decentralgabe) [inline pkg/service/manifest/model/model.go:25]: issue tag plz

File: pkg/service/manifest/model/model.go [hunk 8]
```diff
@@ -142,3 +142,42 @@ type CredentialOverride struct {
...
+type GetRequestResponse struct {
+	ID              string  `json:"id"`
+	ManifestRequest Request `json:"manifestRequest"`
+}
+
+type DeleteRequestRequest struct {
+	ID string `json:"id" validate:"required"`
+}
+
+type ListRequestsRequest struct{}
```
Comment (decentralgabe) [inline pkg/service/manifest/model/model.go:167]: don't need this

File: pkg/service/manifest/model/model.go [hunk 9]
```diff
@@ -142,3 +142,42 @@ type CredentialOverride struct {
...
+type Request struct {
+	common.Request
+
+	// ID of the credential manifest used for this request.
+	ManifestID string `json:"manifestId" validate:"required"`
+
+	// CredentialManifestJWT is a JWT token with a "presentation_definition" claim within it. The
+	// value of the field named "presentation_definition.id" matches PresentationDefinitionID.
+	// This is an output only field.
+	CredentialManifestJWT keyaccess.JWT `json:"CredentialManifestJwt"`
```
Comment (decentralgabe) [inline pkg/service/manifest/model/model.go:182]:
```suggestion
	CredentialManifestJWT keyaccess.JWT `json:"credentialManifestJwt"`
```

**Review by decentralgabe — APPROVED** [review]
lgtm few small comments

File: pkg/service/common/request.go (hunk 3 above)
Comment (andresuribe87) [reply pkg/service/common/request.go]: Done

File: pkg/service/common/request.go (hunk 4 above)
Comment (andresuribe87) [reply pkg/service/common/request.go]: Done

File: pkg/server/router/manifest.go (hunk 1 above)
Comment (andresuribe87) [reply pkg/server/router/manifest.go]: Done

**Pull request merged**